1.5/Source/I_Aint_Building_That/Settings.cs

```diff
@@ -14,6 +16,12 @@ public class Settings : ModSettings
 		private Vector2 _scrollPosition = Vector2.zero;
 		private string _searchQuery = string.Empty;
 		private Dictionary<string, Lazy<BuildableDef>> cachedDefs = new();
+		public List<Ability> HiddenAbilities = [];
```

Comment:
This needs to be saved in ExposeData or the hides won't persist through restarts.

I'm not sure it's possible to save the ability objects directly though as the referenced pawns will only exist when loaded into a save so I'm not sure what'd happen opening the settings from the main menu. Can probably be worked around by something like storing the abilityId instead of the whole thing only looking up the full object for use in the settings page. I think the ideal might be storing a little struct of pawn name, ability label, id or something so the interface can still work even if the pawn is dead, or not in the current save or whatever.

Could also just hide the abilities tab unless you're in game but either way, needs to not crash trying to load in the saved list.

Reply:
As you're hiding at the Pawn+Ability level instead of pure Def level you may end up with saved abilities for dead pawns. I don't think we need to be so bothered about removing them immediately on death but probably as a one-off on game loading or something clearing any abilities for dead pawns feels necessary.

Reply:
If you hide it at def level, then ability will disappear from all pawns (maybe even from NPCs).
I don't know how to save it. Most ExposeData guides are outdated, I don't know how to use it and when to use it ><